SSLPeerUnverifiedException: No peer certificate in PrefixUpdater.java

By Ramnarayan.K 	

User comment:
Have had 3 failures in 14 uploads or attempts to upload,  usually when it is searching for tags

```
USER_COMMENT=Searching for tags while uploading an image 
ANDROID_VERSION=4.4.4
APP_VERSION_NAME=1.12
BRAND=Xiaomi
PHONE_MODEL=2014818
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: An error occured while executing doInBackground()
at android.os.AsyncTask$3.done(AsyncTask.java:300)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
at java.util.concurrent.FutureTask.run(FutureTask.java:242)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:841)
Caused by: java.lang.RuntimeException: javax.net.ssl.SSLPeerUnverifiedException: No peer certificate
at fr.free.nrw.commons.category.PrefixUpdater.doInBackground(PrefixUpdater.java:66)
at fr.free.nrw.commons.category.CategorizationFragment$4.doInBackground(CategorizationFragment.java:366)
at fr.free.nrw.commons.category.CategorizationFragment$4.doInBackground(CategorizationFragment.java:361)
at android.os.AsyncTask$2.call(AsyncTask.java:288)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
... 3 more
Caused by: javax.net.ssl.SSLPeerUnverifiedException: No peer certificate
at com.android.org.conscrypt.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:146)
at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:93)
at org.apache.http.conn.ssl.SSLSocketFactory.createSocket(SSLSocketFactory.java:388)
at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:165)
at org.apache.http.impl.conn.AbstractPoolEntry.open(AbstractPoolEntry.java:164)
at org.apache.http.impl.conn.AbstractPooledConnAdapter.open(AbstractPooledConnAdapter.java:119)
at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:360)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:555)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:487)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:465)
at in.yuvi.http.fluent.Http$HttpRequestBuilder.asResponse(Http.java:470)
at org.mediawiki.api.ApiResult.fromRequestBuilder(ApiResult.java:44)
at org.mediawiki.api.MWApi.makeRequest(MWApi.java:186)
at org.mediawiki.api.MWApi.access$000(MWApi.java:30)
at org.mediawiki.api.MWApi$RequestBuilder.get(MWApi.java:46)
at fr.free.nrw.commons.category.PrefixUpdater.doInBackground(PrefixUpdater.java:63)
... 7 more
java.lang.RuntimeException: javax.net.ssl.SSLPeerUnverifiedException: No peer certificate
at fr.free.nrw.commons.category.PrefixUpdater.doInBackground(PrefixUpdater.java:66)
at fr.free.nrw.commons.category.CategorizationFragment$4.doInBackground(CategorizationFragment.java:366)
at fr.free.nrw.commons.category.CategorizationFragment$4.doInBackground(CategorizationFragment.java:361)
at android.os.AsyncTask$2.call(AsyncTask.java:288)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:841)
Caused by: javax.net.ssl.SSLPeerUnverifiedException: No peer certificate
at com.android.org.conscrypt.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:146)
at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:93)
at org.apache.http.conn.ssl.SSLSocketFactory.createSocket(SSLSocketFactory.java:388)
at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:165)
at org.apache.http.impl.conn.AbstractPoolEntry.open(AbstractPoolEntry.java:164)
at org.apache.http.impl.conn.AbstractPooledConnAdapter.open(AbstractPooledConnAdapter.java:119)
at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:360)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:555)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:487)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:465)
at in.yuvi.http.fluent.Http$HttpRequestBuilder.asResponse(Http.java:470)
at org.mediawiki.api.ApiResult.fromRequestBuilder(ApiResult.java:44)
at org.mediawiki.api.MWApi.makeRequest(MWApi.java:186)
at org.mediawiki.api.MWApi.access$000(MWApi.java:30)
at org.mediawiki.api.MWApi$RequestBuilder.get(MWApi.java:46)
at fr.free.nrw.commons.category.PrefixUpdater.doInBackground(PrefixUpdater.java:63)
... 7 more
javax.net.ssl.SSLPeerUnverifiedException: No peer certificate
at com.android.org.conscrypt.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:146)
at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:93)
at org.apache.http.conn.ssl.SSLSocketFactory.createSocket(SSLSocketFactory.java:388)
at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:165)
at org.apache.http.impl.conn.AbstractPoolEntry.open(AbstractPoolEntry.java:164)
at org.apache.http.impl.conn.AbstractPooledConnAdapter.open(AbstractPooledConnAdapter.java:119)
at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:360)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:555)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:487)
at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:465)
at in.yuvi.http.fluent.Http$HttpRequestBuilder.asResponse(Http.java:470)
at org.mediawiki.api.ApiResult.fromRequestBuilder(ApiResult.java:44)
at org.mediawiki.api.MWApi.makeRequest(MWApi.java:186)
at org.mediawiki.api.MWApi.access$000(MWApi.java:30)
at org.mediawiki.api.MWApi$RequestBuilder.get(MWApi.java:46)
at fr.free.nrw.commons.category.PrefixUpdater.doInBackground(PrefixUpdater.java:63)
at fr.free.nrw.commons.category.CategorizationFragment$4.doInBackground(CategorizationFragment.java:366)
at fr.free.nrw.commons.category.CategorizationFragment$4.doInBackground(CategorizationFragment.java:361)
at android.os.AsyncTask$2.call(AsyncTask.java:288)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:841)
```

--- a/app/src/main/java/fr/free/nrw/commons/LicenseList.java
+++ b/app/src/main/java/fr/free/nrw/commons/LicenseList.java
@@ -64,14 +64,14 @@ public class LicenseList {
     }
 
     public String nameForTemplate(String template) {
-        Log.d("Commons", "LicenseList.nameForTemplate: template: " + template);
+        //Log.d("Commons", "LicenseList.nameForTemplate: template: " + template);
         String stringId = nameIdForTemplate(template);
-        Log.d("Commons", "LicenseList.nameForTemplate: stringId: " + stringId);
+        //Log.d("Commons", "LicenseList.nameForTemplate: stringId: " + stringId);
         int nameId = stringIdByName(stringId);
-        Log.d("Commons", "LicenseList.nameForTemplate: nameId: " + nameId);
+        //Log.d("Commons", "LicenseList.nameForTemplate: nameId: " + nameId);
         if(nameId != 0) {
             String name = res.getString(nameId);
-            Log.d("Commons", "LicenseList.nameForTemplate: name: " + name);
+            //Log.d("Commons", "LicenseList.nameForTemplate: name: " + name);
             return name;
         }
         return template;

--- a/app/src/main/java/fr/free/nrw/commons/category/MethodAUpdater.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/MethodAUpdater.java
@@ -67,6 +67,7 @@ public class MethodAUpdater extends AsyncTask<Void, Void, ArrayList<String>> {
                     .get();
             Log.d(TAG, "Method A URL filter" + result.toString());
         } catch (IOException e) {
+            Log.e(TAG, "IO Exception: ", e);
             //Return empty arraylist
             return categories;
         }

--- a/app/src/main/java/fr/free/nrw/commons/category/MethodAUpdater.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/MethodAUpdater.java
@@ -67,7 +67,8 @@ public class MethodAUpdater extends AsyncTask<Void, Void, ArrayList<String>> {
                     .get();
             Log.d(TAG, "Method A URL filter" + result.toString());
         } catch (IOException e) {
-            throw new RuntimeException(e);
+            //Return empty arraylist
+            return categories;
         }
 
         ArrayList<ApiResult> categoryNodes = result.getNodes("/api/query/search/p/@title");

--- a/app/src/main/java/fr/free/nrw/commons/category/PrefixUpdater.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/PrefixUpdater.java
@@ -11,6 +11,8 @@ import org.mediawiki.api.MWApi;
 import java.io.IOException;
 import java.util.ArrayList;
 
+import javax.net.ssl.SSLPeerUnverifiedException;
+
 import fr.free.nrw.commons.CommonsApplication;
 
 /**
@@ -63,7 +65,8 @@ public class PrefixUpdater extends AsyncTask<Void, Void, ArrayList<String>> {
                     .get();
             Log.d(TAG, "Prefix URL filter" + result.toString());
         } catch (IOException e) {
-            throw new RuntimeException(e);
+            //Return empty arraylist
+            return categories;
         }
 
         ArrayList<ApiResult> categoryNodes = result.getNodes("/api/query/allcategories/c");

--- a/app/src/main/java/fr/free/nrw/commons/category/PrefixUpdater.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/PrefixUpdater.java
@@ -65,6 +65,7 @@ public class PrefixUpdater extends AsyncTask<Void, Void, ArrayList<String>> {
                     .get();
             Log.d(TAG, "Prefix URL filter" + result.toString());
         } catch (IOException e) {
+            Log.e(TAG, "IO Exception: ", e);
             //Return empty arraylist
             return categories;
         }

